org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/FDialog.java

```diff
@@ -423,6 +423,33 @@ public void onCallback(Object result) {
         Messagebox.showDialog(s, Util.isEmpty(title) ? AEnv.getDialogHeader(Env.getCtx(), windowNo) : title,
         		Messagebox.OK | Messagebox.INPUT, Messagebox.QUESTION, weditor, msgCallback, (msgCallback == null));
     }
+    
+    /**
+     * Confirmation dialog before deleting the records. 
+     * @param windowNo
+     * @param weditor
+     * @param adMessage
+     * @param adMessageArgs
+     * @param title
+     * @param correctInput
+     * @param callback
+     */
+    public static void askForInputDeleteConfirmation(int windowNo, WEditor weditor, String adMessage, Object[] adMessageArgs, String title, String correctInput, final Callback<Object> callback)
```

Comment:
instead of passing correctInput, a more general approach is to create a validation interface instead. that should provide support for different type of editor too.